Formula/git.rb

```diff
@@ -25,11 +25,11 @@ class Git < Formula
 
   uses_from_macos "curl"
   uses_from_macos "expat"
-  uses_from_macos "openssl"
   uses_from_macos "zlib"
 
   on_linux do
     depends_on "linux-headers@4.4"
+    depends_on "openssl@1.1" # Uses CommonCrypto on macOS
```

Comment:
We'll need to ship new bottles when `openssl@1.1` becomes keg-only, I think. So we may want to do this here.

Reply:
If we need a mass rebottle then I imagine this will be merged before `openssl@1.1` becomes keg-only.

I'm really hoping we don't need a mass rebottle.

Reply:
We can probably avoid a mass rebottle by doing a Linux-only post-install where we fix up the formula `RPATH`s. Or we could carry something like it around in `brew`.

Reply:
If necessary, we can keg only `openssl@3` until we sort thing out which will unblock things.

But we really should be treating every dependency as if it was keg only. This is not the first time we've done an update like this, and it won't be the last.

Reply:
Please do not hurry to switch to OpenSSL v3: There is an issue that will brick older Macs with this: https://github.com/openssl/openssl/issues/16670

Thank you